runtime/src/anchor/mod.rs

```diff
@@ -191,15 +188,15 @@ decl_module! {
             <fees::Module<T>>::pay_fee_to_author(who, fee)?;
 
             let block_num = <frame_system::Module<T>>::block_number();
-            let child_storage_key = common::generate_child_storage_key(stored_until_date_from_epoch);
+            let child_info = common::generate_child_storage_key(stored_until_date_from_epoch);
```

Comment:
idea was to bucket a chunk of anchors under a timestamp. This will be easier to evict the anchors

Reply:
Agree with @mikiquantum's point here – if we don't use a prefix here, there might be an easy collision since timestamps alone are not really unique enough. Does that make sense?

Reply:
Timestamps here are the bucket for anchors and AnchorID is really unique. How is using a prefix provides any uniqueness here?

A timestamp provides same uniqueness as constant-prefix + epoch
Anchor ID is the one that provides the system-wide uniqueness. 

We use the timestamp to bundle a bunch of anchorIDs like this
(evict_date_since_epoch) =>[
0x... // anchorID
0x...
0x...
]
Here `evict_date_since_epoch` date is rounded till per day so at a given day, all the anchors that were created on that day will fall under this bucket. We did this so that any specific anchor buckets will be evicted as easy as killing `evict_date_since_epoch` there by eliminating all the anchors under it.

Reply:
If we kill by prefix, and the prefix is just the timestamp without a namespace, there is some risk that we have a collision with keys that where also created at that timestamp, right?

We have no other logic in here right now that uses timestamps as keys, but we (or Substrate) might introduce timestamp keys in the future. With no namespace prefix, there will be collisions IIUC.

Reply:
> there is some risk that we have a collision with keys that where also created at that timestamp, right?
true. That is the reason why we need to figure out the how to use the existing keyspace `ANCHOR` with new childtrie APIs. is that a simple prefix? I do not know ATM until we understand the how the keyspace was used earlier which we are discussing with parity ATM as mentioned in the above comment

Reply:
> Does it mean we need to add some prefix to this epoch value? Since a timestamp is rather generic.

@mikiquantum I do not know how we can bring back the prefix `ANCHOR` without breaking any storage. I will let you know as soon as I have more info but I can see what you mean by being that generic